src/NLog.Targets.Syslog/SyslogTarget.cs

```diff
@@ -75,35 +75,47 @@ protected override void FlushAsync(AsyncContinuation asyncContinuation)
             Task.WhenAll(tasks)
                 .ContinueWith(t =>
                 {
-                    InternalLogger.Debug("[Syslog] Explicit flush completed");
+                    InternalLogger.Debug(t.Exception?.GetBaseException(), "[Syslog] Explicit flush completed");
                     asyncContinuation(t.Exception?.GetBaseException());
-                })
-                .Wait();
+                });
+        }
+
+        /// <inheritdoc />
+        protected override void CloseTarget()
+        {
+            DisposeDependencies();
+            base.CloseTarget();
         }
 
         /// <summary>Disposes the instance</summary>
         protected override void Dispose(bool disposing)
         {
             if (disposing)
             {
-                Enforcement.PropertyChanged -= Init;
-                MessageCreation.PropertyChanged -= Init;
-                MessageSend.PropertyChanged -= Init;
```

Comment:
Moving this to DisposeDependencies prevents event handlers from working on a reinitialization (see Init method)

Reply:
Dispose is only called when `NLog.LogManager.Shutdown()` is called.

Reply:
I was referring to [this](https://github.com/luigiberrettini/NLog.Targets.Syslog/blob/4a2bc0faecffcd7ca3fa4dea1b138e490894b525/src/NLog.Targets.Syslog/SyslogTarget.cs#L52)

Then for sure the CloseTarget should be fixed, but this is due to not properly implementing the Dispose pattern so that every target implementer has to know the internals of NLog

Reply:
When making extensions for a framework (Like NLog), then it is good to know the framework rules. Like the rules of object-lifetime. InitializeTarget matching CloseTarget and Dispose matching Constructor.

Reply:
And when making a framework it is good to use standard practices and patterns and conventions in the language of choice so that people extending it do not have to study for things that should be standard.
Since the beginning the Dispose pattern documentation advised to have Dispose and Close methods and the Close calling the Dispose (see File, DB connections, etc.).
Moreover if, to work properly, some methods should be implemented, an abstract class or an interface should be used, so that, again, standard programming knowledge and hints at how something should be used are given by OOP and the language

Reply:
When inside `InitializeTarget()` then `IsInitialized` is always false, so the if-statement will never trigger the call to `DisposeDependencies`

Reply:
@luigiberrettini I'm sorry to disappoint you. But I'm not the architect behind the NLog-object-lifetime-logic. I'm just trying to convey how it is currently implemented. I can only agree that NLog has earned some warts by being an old framework (Just like everything in .Net framework is not always logical). But one thing I can do for you is update the wiki-page:

https://github.com/NLog/NLog/wiki/How-to-write-a-custom-target#how-to-handle-object-lifetime

Reply:
> When inside `InitializeTarget()` then `IsInitialized` is always false, so the if-statement will never trigger the call to `DisposeDependencies`

`IsInitialized` is false during `InitializeTarget()` but not when a `PropertyChanged` event triggers a reinitialization for which I did not want to unsubscribe and subscribe event handlers

Reply:
Guess you should add a comment about having this special reinit-logic, since it diverts from the default NLog object lifetime.

Reply:
Does reconfiguration triggers `InitializeTarget` and/or `CloseTarget`?

Reply:
Yes if you re-assign the same config like this:
```c#
var config = LogManager.Configuration;
LogManager.Configuration = config;
```
Can also sometimes happen when you do this:
```
LogManager.Configuration = LogManager.Configuration?.Reload();
```